File: subarrsum.cpp
#include <array>
using namespace std;		
#include<iostream>
#include <string> 
#include <climits>
#include <stack>
#include <vector>
#include <map>
//c++ subarrsum.cpp -o executable.exe & executable
// array<int, 5> myarray = { 1, 2, 3, 4, 5 };
int subarraySum(vector<int>nums, int k) {
    int count = 0;
    int sum = 0;
    int *b = &sum;
    cout << *b << endl;
    map<int, int> m;
    if(nums.size()==0) return 0;
    for(int i=0; i<nums.size(); i++){
        sum+=nums[i];
        if (m.find(sum-k) != m.end()) count += m[sum-k];
        if (m.find(sum) != m.end()) m[sum] = m[sum]+1;
        else m[sum] = 1;
    }
    return count;
};
int main()
{
    vector<int> nums={1,1,1};
    int k=2;
    subarraySum(nums, k);
    return 0; 
}

File: subarrsum.js
var subarraySum = function(nums, k) {
    let count = 0;
    let sum = 0;
    let map = new Map();
    map.set(0,1);
    if(nums.length==0) return 0;
    for(let i=0; i<nums.length; i++){
        sum+=nums[i]
        if (map.has(sum-k)) count += map.get(sum-k);
        if (map.has(sum)) map.set(sum, map.get(sum)+1);
        else map.set(sum, 1);
    }
    return count;
}
console.log(subarraySum([1,1,1], 2))